src/Microsoft.FeatureManagement.AspNetCore/TargetingHttpContextMiddleware.cs

```diff
@@ -50,14 +53,15 @@ public async Task InvokeAsync(HttpContext context, ITargetingContextAccessor tar
 
             if (targetingContext != null)
             {
-                context.Items[TargetingIdKey] = targetingContext.UserId;
+                var activityFeature = httpContext.Features.Get<IHttpActivityFeature>();
+                activityFeature?.Activity?.AddBaggage(TargetingIdKey, targetingContext.UserId);
```

Comment:
The null check on activity feature is a safeguard, but it's still unexpected right? I'm thinking we should log a warning if it's null.

Reply:
Adjusted to:

```csharp
var activityFeature = httpContext.Features.Get<IHttpActivityFeature>();

if (activityFeature == null)
{
    _logger.LogDebug("The IHttpActivityFeature from the IFeatureCollection was null");
}
else if (activityFeature.Activity == null)
{
    _logger.LogDebug("The Activity on the IHttpActivityFeature was null");
} else
{
    activityFeature.Activity.AddBaggage(TargetingIdKey, targetingContext.UserId);
}
```

Reply:
That's a debug log. You don't consider it worth a warning?

Reply:
I was following suit on the other null check that causes the middleware to no op:
```csharp
    _logger.LogDebug("The targeting context accessor returned a null TargetingContext");
```

But I suppose these two are more unexpected than that one?

Reply:
> But I suppose these two are more unexpected than that one?

Yes, I would say that. And also I believe we didn't want to log warning in the middleware for not having a targeting context because the idea was to consider the middleware may be in use even if targeting is not enabled.

Reply:
PR: https://github.com/microsoft/FeatureManagement-Dotnet/pull/471